src/net_processing.cpp

```diff
@@ -3531,7 +3531,9 @@ void ProcessMessage(
         if (pfrom.HasPermission(PF_ADDR)) {
             vAddr = connman.GetAddresses();
         } else {
-            vAddr = connman.GetAddresses(pfrom.addr.GetNetwork());
+            LOCK(pfrom.cs_hSocket);
```

Comment:
This seems like a slight layer violation, with network and socket details bleeding up into net_processing. Did you consider just passing a `CNode&` reference down to CConnMan and allowing it to get the bind address and network?